packages/x-data-grid/src/material/index.tsx

```diff
@@ -303,6 +242,7 @@ function BaseCheckbox(props: GridSlotProps['baseCheckbox'], ref: React.Ref<HTMLB
     return (
       <Checkbox
         {...other}
+        {...material}
         className={className}
         inputProps={slotProps?.htmlInput}
         ref={handleRef}
```

Comment:
Shouldn't we merge `className` with `material.className`? Same for `inputProps`

Reply:
It makes the bindings more verbose, and I don't like having multiple ways to do one thing. I've removed `slotProps` and `className` from material props, those are already usable through the base components API anyway.

Reply:
Done for `className`, but for `slotProps` I don't think we should start combining all the options (`inputProps` is also an alias for `slotProps.input`), it makes the bindings more verbose and users would be using either one of those props, not both.